### DISCUSSION TITLE
How to access run target arguments?

### DISCUSSION THREAD

**Discussion by dkaszews** [xmake-io/xmake#4939] [Q&A]
`xmake run --help` shows that you can pass arguments to target:

```
Usage: $xmake run|r [options] [target] [arguments]
          arguments ...              The target arguments
```

but I cannot figure out how to access them in `on_run`. Documentation does not mention them and nothing like `os.args()` or `target.arguments()` works.

I am using phony targets as helpers to run test executables with some default arguments, would like to use this feature to add some extra ones from command line.

**Comment by waruqi** [comment]
see https://github.com/xmake-io/xmake/blob/537971777a7b6086546d234ea9f5975464903c7e/xmake/actions/run/main.lua#L51
